### PR TITLE
Chef-solo: Requesting fix for windows

### PR DESCRIPTION
**Pull request by DonCherryMaraschino**
Great code here, would like to have it pulled into master so that we could all use it.

### CODE REVIEW

File: provisioner/chef-solo/provisioner.go [hunk 1]
```diff
@@ -417,20 +433,31 @@ func (p *Provisioner) createJson(ui packer.Ui, comm packer.Communicator) (string
...
+		cmd := &packer.RemoteCmd{
+			Command: fmt.Sprintf("mkdir -p '%s'", dir),
+		}
+		if err := cmd.StartWithUi(comm, ui); err != nil {
+			return err
+		}
+		if cmd.ExitStatus != 0 {
+			return fmt.Errorf("Non-zero exit status.")
+		}
+	}else {
```
Comment (LinusU) [inline provisioner/chef-solo/provisioner.go]: Missing space

**Comment by cbednarski** [comment]
Thanks for the PR, and sorry to keep you waiting so long for a review. I think this conflicts with #1215. I will need to re-evaluate this PR after merging the other one. From a glance it looks like the functionality is not duplicated in the other PR but I'm certain there will be a merge conflict for me to resolve.

**Comment by mwhooker** [comment]
It looks like the majority of this work is already in master. However, I still think we're constructing paths incorrectly. Let me try to use `filepath.Join` and throw up a PR

**Comment by mwhooker** [comment]
I'm curious if this is still an issue. It looks like there's logic for abstracting windows guests

https://github.com/mitchellh/packer/blob/master/provisioner/chef-solo/provisioner.go#L34-L37
https://github.com/mitchellh/packer/blob/master/provisioner/guest_commands.go#L24-L27

we still construct paths using `/` literals (https://github.com/mitchellh/packer/blob/master/provisioner/chef-solo/provisioner.go#L243)
however those paths get written to `config.rb`, where it comes ruby's job to interpret the paths properly. It also looks like ruby doesn't care about path separators (at least according to http://stackoverflow.com/a/7173356 . I'm no ruby expert)

I'm going to close this for now, but please do re-open if there are still problems this solves

**Pull request closed without merging**